corehq/apps/export/forms.py

```diff
@@ -451,14 +459,46 @@ def get_case_filter(self):
                                     groups=case_sharing_groups)
 
 
-class FilterCaseESExportDownloadForm(BaseFilterExportDownloadForm):
+class FilterCaseESExportDownloadForm(GenericFilterCaseExportDownloadForm):
     _export_type = 'case'
 
+    date_range = forms.CharField(
+        label=ugettext_lazy("Date Range"),
+        required=True,
+        widget=DateRangePickerWidget(),
+        help_text="Export cases modified in this date range",
+    )
+
+    def __init__(self, domain_object, timezone, *args, **kwargs):
+        self.timezone = timezone
+        super(FilterCaseESExportDownloadForm, self).__init__(domain_object, timezone, *args, **kwargs)
+
+        # update date_range filter's initial values to span the entirety of
+        # the domain's submission range
+        default_datespan = datespan_from_beginning(self.domain_object, self.timezone)
+        self.fields['date_range'].widget = DateRangePickerWidget(
+            default_datespan=default_datespan
+        )
+
+
     def get_edit_url(self, export):
         from corehq.apps.export.views import EditNewCustomCaseExportView
         return reverse(EditNewCustomCaseExportView.urlname,
                        args=(self.domain_object.name, export.get_id))
 
+    def _get_datespan(self):
+        date_range = self.cleaned_data['date_range']
+        dates = date_range.split(DateRangePickerWidget.separator)
+        startdate = dateutil.parser.parse(dates[0])
+        enddate = dateutil.parser.parse(dates[1])
+        return DateSpan(startdate, enddate)
+
+    def _get_datespan_filter(self):
+        datespan = self._get_datespan()
+        if datespan.is_valid():
+            datespan.set_timezone(self.timezone)
+            return ModifiedOnRangeFilter(gte=datespan.startdate, lt=datespan.enddate + timedelta(days=1))
```

Comment:
`lt` is non inclusive, so make sure that it includes the day on the end date. I don't think we can use `lte` because ES searches with date and times so this is `start_date_at_midnight` until `end_date_at_midnight` The form filter uses this same pattern